charts/apisix/values.yaml

```diff
@@ -120,6 +120,14 @@ apisix:
 nameOverride: ""
 fullnameOverride: ""
 
+serviceAccount:
+  create: false  # Set to `true` to create ServiceAccount for Kubernetes service discovery.
+  annotations: {}
+  name: ""
+
+rbac:
+  create: false  # Set to `true` to create RBAC resources for Kubernetes service discovery.
```

Comment:
Since the configuration content here is actually tied to a specific feature of APISIX, I recommend moving it to a section. It does not have the same meaning or behavior as the configuration item of the same name in most Helm charts

For example put it under `discovery.registry.kubernetes` .

Reply:
I think `serviceAccount` can stay at the same section, as its behavior is the same as in other Helm charts. The comment in line 124 can be removed, since it is not strongly tied to this Kubernetes service discovery feature now; and I might need to update the comment on configuring `discovery.registry.kubernetes`.

As for the `rbac`, is moving it to, like `discovery.kubernetesRegistryRBAC` better? I noticed [here](https://github.com/apache/apisix-helm-chart/blob/master/charts/apisix/templates/configmap.yaml#L244) the content of `discovery.registry.*` is directly render as part of APISIX config.

Reply:
> I think `serviceAccount` can stay at the same location, as its behavior is the same as in other Helm charts.

ok


----

Another important thing comes to my mind, maybe I'm just bothered by your title and comments.

We can simply align with other popular Charts.
Keep the following fields

* serviceAccount
* rbac

But don't associate them with APISIX's kubernetes discovery


Then we need a configuration for RBAC that handles kubernetes discovery specifically

Reply:
I see, so I change the title to "support ServiceAccount and RBAC"?

Reply:
And probably remove the comments at `serviceAccount` and `rbac`?

Reply:
right! thanks

Reply:
I've updated accordingly, thanks for the recommendation!